app/models/rejection.rb

```diff
@@ -37,6 +37,9 @@ class Rejection < ActiveRecord::Base
   validates :reasons, presence: true
   validate :validate_allowance_renews_on_date
 
+  before_save do
```

Comment:
How do you see it impacting the aggregation calculations? Those are done in sql so it would be unchanged if I remember correctly.

I personally see more downsides by not keeping the object consistent during the whole lifecycle, however if it has a big knock on effect introducing this change lets keep it as it is.